torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -2333,6 +2334,65 @@ Tensor eig_backward(const std::vector<torch::autograd::Variable> &grads, const T
   return at::linalg_solve(Uh, at::matmul(U_contrib, Uh) + D_contrib * Uh);
 }
 
+Tensor linalg_eig_backward(const std::vector<torch::autograd::Variable> &grads,
+                           const Tensor& self,
+                           const Tensor& L,
+                           const Tensor& V) {
+  // https://arxiv.org/pdf/1701.00392.pdf Eq 4.77
+  // For A = VLV^{-1}, denoting the gradients gA, gV and gL, we have
+  // gA = V^{-H}(diag_embed(gL) + (V^H gV -V^HV diag(real(V^H gV))) / E*)V^H
+  // Where:
```

Comment:
I would add the original comments from `eig_backward` regarding why we need the projection which is different from what the references considers (i.e. the eigenvectors normalized to have 1 norm).

Reply:
I will add them. Even then, note that I also changed the reference for one that takes in account that point and makes a derivation analogous to the one you made.

Reply:
Even better, as discussed offline, I think it'd be good to put together a nice formal document where we derive all these things and just link all of them to that document. This is something that we can work on in the future.